addons/account/models/partner.py

```diff
@@ -209,6 +209,10 @@ def _get_fpos_by_region(self, country_id=False, state_id=False, zipcode=False, v
             fpos = self.search(base_domain + null_country_dom, limit=1)
         return fpos
 
+    def _get_vat_required(self, delivery, company=None):
+        """ Hook for determining VAT validity with more complex VAT requirements """
+        return bool(delivery.vat)
```

Comment:
Is the VAT the same on the partner and the delivery address?

Reply:
Generally yes. I think it's the function `_fields_sync` in base that sets the vat to be the same across the child (in this case delivery) partners of a given partner.